src/datasets/io/json.py

```diff
@@ -63,49 +64,84 @@ def __init__(
+            key=slice(offset, offset + self.batch_size),
+            indices=self.dataset._indices,
+        )
+        var = batch.to_pandas().to_json(path_or_buf=None, orient=orient, lines=lines, **self.to_json_kwargs)
+        return var.encode(self.encoding)
+
     def _write(
         self,
         file_obj: BinaryIO,
-        batch_size: int,
-        encoding: str = "utf-8",
-        orient="records",
-        lines=True,
+        orient,
+        lines,
         **to_json_kwargs,
     ) -> int:
         """Writes the pyarrow table as JSON lines to a binary file handle.
 
         Caller is responsible for opening and closing the handle.
         """
         written = 0
-        _ = to_json_kwargs.pop("path_or_buf", None)
-
-        for offset in utils.tqdm(
-            range(0, len(self.dataset), batch_size), unit="ba", disable=bool(logging.get_verbosity() == logging.NOTSET)
-        ):
-            batch = query_table(
-                table=self.dataset.data,
-                key=slice(offset, offset + batch_size),
-                indices=self.dataset._indices if self.dataset._indices is not None else None,
-            )
-            json_str = batch.to_pandas().to_json(path_or_buf=None, orient=orient, lines=lines, **to_json_kwargs)
-            if not json_str.endswith("\n"):
-                json_str += "\n"
-            written += file_obj.write(json_str.encode(encoding))
+
+        if self.num_proc is None or self.num_proc == 1:
+            for offset in utils.tqdm(
+                range(0, len(self.dataset), self.batch_size),
+                unit="ba",
+                disable=bool(logging.get_verbosity() == logging.NOTSET),
+                desc="Creating json from Arrow format",
+            ):
+                batch = query_table(
+                    table=self.dataset.data,
+                    key=slice(offset, offset + self.batch_size),
+                    indices=self.dataset._indices if self.dataset._indices is not None else None,
+                )
+                json_str = batch.to_pandas().to_json(path_or_buf=None, orient=orient, lines=lines, **to_json_kwargs)
+                if not json_str.endswith("\n"):
+                    json_str += "\n"
+                written += file_obj.write(json_str.encode(self.encoding))
+        else:
+            pool = multiprocessing.Pool(processes=self.num_proc)
+
+            for json_str in utils.tqdm(
+                pool.imap(
+                    self._batch_json,
+                    [(offset, orient, lines) for offset in range(0, len(self.dataset), self.batch_size)],
+                ),
+                total=(len(self.dataset) // self.batch_size) + 1,
+                unit="ba",
+                disable=bool(logging.get_verbosity() == logging.NOTSET),
+                desc="Creating json from Arrow format",
+            ):
+                written += file_obj.write((json_str))
+
+            pool.close()
+            pool.join()
```

Comment:
`multiprocessing.Pool` can be used as a context managers, and it's a good practice to use context managers whenever possible for better error handling:
```suggestion
            with multiprocessing.Pool(self.num_proc) as pool:
                for json_str in utils.tqdm(
                    pool.imap(
                        self._batch_json,
                        [(offset, orient, lines) for offset in range(0, len(self.dataset), self.batch_size)],
                    ),
                    total=(len(self.dataset) // self.batch_size) + 1,
                    unit="ba",
                    disable=bool(logging.get_verbosity() == logging.NOTSET),
                    desc="Creating json from Arrow format",
                ):
                    written += file_obj.write((json_str))

                pool.close()
                pool.join()
```

Reply:
Thanks for the suggestion, @mariosasko! (TIL!)